plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/labeling/helper/StyledTextCalculationHelper.xtend

```diff
@@ -145,50 +153,64 @@ class StyledTextCalculationHelper {
+			return dispatchGetStyledText(n4GetterDeclaration.definedGetter);
+	
+		return getLabelProvider.getSuperStyledText(n4GetterDeclaration);
+
+	}
+
+	/**
+	 * Returns the styled text for the given getter, does not accesses the AST.
+	 */
+	def dispatch StyledString dispatchGetStyledText(TGetter tgetter) {
+		var styledText = getLabelProvider.getSuperStyledText(tgetter);
 		var typeStr = "";
 
-		if (definedGetter !== null) {
-			if (definedGetter.declaredTypeRef !== null)
-				typeStr = " : " + getTypeRefDescription(definedGetter.declaredTypeRef)
+		if (tgetter.declaredTypeRef !== null)
+			typeStr = " : " + getTypeRefDescription(tgetter.declaredTypeRef)
 
-			if (n4GetterDeclaration.declaredTypeRef === null) {
-				styledText = styledText.append(typeStr, INFERRED_TYPE_STYLER)
-			} else {
-				styledText = styledText.append(typeStr)
-			}
-		}
+		styledText = styledText.append(typeStr)
 		return styledText;
 	}
 
 	/** produces e.g. setterName: paramTypeName */
 	def dispatch StyledString dispatchGetStyledText(N4SetterDeclaration n4SetterDeclaration) {
-		var styledText = getLabelProvider.getSuperStyledText(n4SetterDeclaration)
-		val definedSetter = n4SetterDeclaration.definedSetter
-		if (definedSetter !== null) {
-			if (n4SetterDeclaration.fpar !== null && definedSetter.fpar !== null) {
-				styledText.append(" : ").append(
-					getStyledTextForFormalParameter(n4SetterDeclaration.fpar, definedSetter.fpar))
-			}
+		if (n4SetterDeclaration.definedSetter!==null)
+			return dispatchGetStyledText(n4SetterDeclaration.definedSetter);
+		
+		return getLabelProvider.getSuperStyledText(n4SetterDeclaration);
+	}
+
+	/**
+	 * Returns the styled text for the given setter, does not access AST.
+	 */
+	def dispatch StyledString dispatchGetStyledText(TSetter tsetter) {
+		var styledText = getLabelProvider.getSuperStyledText(tsetter)
+		if (tsetter.fpar !== null && tsetter.fpar !== null) {
+			styledText.append(" : ").append(
+				getStyledTextForFormalParameter(tsetter.fpar))
 		}
+
 	}
 
 	// produces e.g. fieldName : fieldTypeName
 	def dispatch StyledString dispatchGetStyledText(N4FieldDeclaration n4FieldDeclaration) {
-		var styledText = getLabelProvider.getSuperStyledText(n4FieldDeclaration);
-		val definedField = n4FieldDeclaration.definedField;
+		if (n4FieldDeclaration.definedField!==null)
+			return dispatchGetStyledText(n4FieldDeclaration.definedField);
+		
+		return getLabelProvider.getSuperStyledText(n4FieldDeclaration);
+	}
+	
+	/**
+	 * Returns the styled text for the given field, does not accesses the AST.
+	 */
+	def dispatch StyledString dispatchGetStyledText(TField tfield) {
+		var styledText = getLabelProvider.getSuperStyledText(tfield);
 		var typeStr = "";
-
-		if (definedField?.typeRef !== null)
-			typeStr = " : " + getTypeRefDescription(definedField.typeRef)
-
-		if (n4FieldDeclaration.declaredTypeRef === null) {
-			// type was inferred
-			styledText = styledText?.append(typeStr, INFERRED_TYPE_STYLER)
```

Comment:
Because I do not want to access the AST. That is, we do not have information about inference state in the TModule. So I decided to drop that style. Eventually it is not important for the developer -- we infer so many things, that this is not that exciting anymore.